clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/consumer/AsyncHandlingAutoCommitKafkaConsumer.java

```diff
@@ -567,11 +569,20 @@ private synchronized void clearObsoleteTopicPartitionOffsets(final Collection<To
      * Checks whether offset commits are currently needed for the partitions of the given topic.
      * <p>
      * This may be used to check whether a topic can safely be deleted if no further incoming records are expected.
+     * <br>
+     * NOTE: This method should only be called after the topic partitions have been assigned to this consumer.
+     * In case assignment isn't completed yet and offsets haven't been fetched, this method will return {@code true}
+     * even if no offsets will need to be committed eventually.
      *
      * @param topic The topic to check.
      * @return {@code true} if offsets need to be committed.
      */
-    public boolean isOffsetsCommitNeededForTopic(final String topic) {
+    public synchronized boolean isOffsetsCommitNeededForTopic(final String topic) {
+        if (offsetsMap.keySet().stream().noneMatch(tp -> tp.topic().equals(topic))) {
+            LOG.info("isOffsetsCommitNeededForTopic: no offsets stored for topic {} (yet); offset commit may be needed after having fetched positions",
```

Comment:
will this happen regularly and often in production? Or is this just a safeguard against the peculiarities of our integration tests where we permanently create and delete tenants/topics? If it is the former, then we should probably use DEBUG level ...